### ISSUE TITLE
[a11y] [Applicant] Handle max number of files being reached for a screen reader user

### ISSUE THREAD

**Issue by tallulahkay** [civiform/civiform#13319]
We don't announce that the max file number has been reached and that the user can no longer upload files. One of the major principles of accessibility is that if there's a change in the page context, we need to let people know.

https://github.com/user-attachments/assets/8f6a6ac2-45c8-4180-971a-b96c590565af

**Comment by rockycodes** [comment]
@melanie-exygy is this working as intended for a disabled element? should screen readers announce a disabled element?

**Comment by rockycodes** [comment]
not launch blocking because already exists as an issue with the current flow

**Comment by melanie-exygy** [comment]
> [@melanie-exygy](https://github.com/melanie-exygy) is this working as intended for a disabled element? should screen readers announce a disabled element?

The issue isn't that there's a disabled element or that the disabled element isn't read by the screen reader.  The issue is that we don't announce that the max file number has been reached and that the user can no longer upload files.  One of the major principles of accessibility is that if there's a change in the page context, we need to let people know.  

I understand that it may not be launch blocking, but I do consider it to be a fairly serious a11y issue, probably P1.

**Comment by rockycodes** [comment]
Ah gotcha! Thanks for explaining. I agree this is high priority.

**Comment by salg-r** [comment]
I can take a look at this, but what would be the best approach? Should we modify the hint above the file picker? Right now it reads "Select one or many files (maximum of N)", even if the limit has been reached.

**Comment by myfriendgu** [comment]
Design added here along with annotation note: https://www.figma.com/design/aHJ6bH2vR3CNK2DcJHAeOj/-Admin-DRAFT--CiviForm--Components--%E2%80%A8Styles--Templates?node-id=4407-1889&t=ryQ1fQgDM3ZB4IKK-1

For more context on the decision, please visit the slack thread: https://civiform.slack.com/archives/C0B1Q113Z4M/p1784034528773179

**Comment by rockycodes** [comment]
@salg-r the translations for the string you added have landed! You should be clear to move forward with this issue now.
